extensions/fs/internal.m

```diff
@@ -1090,6 +1091,34 @@ static int fs_pathFromBookmark(lua_State *L) {
     return 1 ;
 }
 
+/// hs.fs.urlFromPath(path) -> string | nil
+/// Function
+/// Returns the encoded URL from a path.
+///
+/// Parameters:
+///  * path - The path
+///
+/// Returns:
+///  * A string or `nil` if path is invalid.
+static int fs_urlFromPath(lua_State *L) {
+    LuaSkin *skin = [LuaSkin sharedWithState:L] ;
+    [skin checkArgs:LS_TSTRING, LS_TBREAK] ;
+
+    NSString *filePath = [skin toNSObjectAtIndex:1];
+    char *absolutePath = realpath([filePath stringByExpandingTildeInPath].UTF8String, NULL);
+
+    if (!absolutePath) {
+        lua_pushnil(L);
+        return 1;
+    }
+    
+    NSURL *fileURL = [[NSURL alloc] initFileURLWithPath:[NSString stringWithFormat:@"%s", absolutePath]];
```

Comment:
Probably "faster" to use `[NSString stringWithUTF8String:absolutePath]`

Reply:
I tried that but got:

`Implicit conversion from nullable pointer 'NSString * _Nullable' to non-nullable pointer type 'NSString * _Nonnull'`

Reply:
Ok, Apple did get a little more anal about Nullable warnings a few Xcode versions back and (in my opinion) dropped the ball when it came to updating some of their method definitions. Meh, it was just a thought.